smart_card_connector_app/src/chrome-api-jstocxxtest.js

```diff
@@ -162,6 +205,111 @@ goog.exportSymbol('testChromeApiProviderToCpp', {
         'onReleaseContextRequested', /*requestId=*/ 42, BAD_CONTEXT);
     await chrome.smartCardProviderPrivate['reportReleaseContextResult']
         .$waitAndVerify();
-  }
+  },
+
+  // Test ListReaders with no readers attached.
+  'testListReaders_none': async function() {
+    let establishContextResult = EMPTY_CONTEXT_RESULT;
+    expectReportEstablishContext(
+        /*requestId=*/ 123, 'SUCCESS', establishContextResult);
+    expectReportListReaders(/*requestId=*/ 124, [], 'NO_READERS_AVAILABLE');
+    mockControl.$replayAll();
+
+    launchPcscServer(/*initialDevices=*/[]);
+    createChromeApiProvider();
+    mockChromeApi.dispatchEvent(
+        'onEstablishContextRequested', /*requestId=*/ 123);
+    await chrome.smartCardProviderPrivate['reportEstablishContextResult']
+        .$waitAndVerify();
+    mockChromeApi.dispatchEvent(
+        'onListReadersRequested', /*requestId=*/ 124,
+        establishContextResult.sCardContext);
+    await chrome.smartCardProviderPrivate['reportListReadersResult']
+        .$waitAndVerify();
+  },
+
+  // Test that ListReaders requested with already released context returns
+  // INVALID_HANDLE error.
+  'testListReaders_releasedContext': async function() {
+    let establishContextResult = EMPTY_CONTEXT_RESULT;
+    expectReportEstablishContext(
+        /*requestId=*/ 123, 'SUCCESS', establishContextResult);
+    expectReportReleaseContext(/*requestId=*/ 124, 'SUCCESS')
```

Comment:
nit: Terminate the statement with a semicolon (ditto in other places below).

Reply:
thanks;